Packages/com.unity.inputsystem/InputSystem/Plugins/iOS/GameController.cs

```diff
@@ -0,0 +1,83 @@
+using System.Runtime.InteropServices;
+using ISX.Controls;
+using ISX.LowLevel;
+using ISX.Utilities;
+using UnityEngine;
+
+namespace ISX.iOS
+{
+    [StructLayout(LayoutKind.Sequential)]
+    public unsafe struct GameControllerState : IInputStateTypeInfo
+    {
+        enum Button 
+        {
+            DpadUp,
+            DpadDown,
+            DpadLeft,
+            DpadRight,
+            LeftStick,
+            RightStick,
+            LeftShoulder,
+            RightShoulder,
+            LeftTrigger,
+            RightTrigger,
+            X,
+            Y,
+            A,
+            B
+        };
+
+        enum Axis 
+        {
+            LeftStickX,
+            LeftStickY,
+            RightStickX,
+            RightStickY
+        };
+
+        public static FourCC kFormat = new FourCC('I', 'G', 'C', ' ');
+        public const int kMaxButtons = 14;
+        public const int kMaxAxis = 4;
+        
+        [InputControl(name = "dpad", template = "Dpad")]
+        [InputControl(name = "dpad/up", template = "Button", bit = (uint)Button.DpadUp)]
+        [InputControl(name = "dpad/right", template = "Button", bit = (uint)Button.DpadRight)]
+        [InputControl(name = "dpad/down", template = "Button", bit = (uint)Button.DpadDown)]
+        [InputControl(name = "dpad/left", template = "Button", bit = (uint)Button.DpadLeft)]  
+        [InputControl(name = "buttonSouth", template = "Button", bit = (uint)Button.A)]
+        [InputControl(name = "buttonWest", template = "Button", bit = (uint)Button.X)]
+        [InputControl(name = "buttonNorth", template = "Button", bit = (uint)Button.Y)]
+        [InputControl(name = "buttonEast", template = "Button", bit = (uint)Button.B)]
+        [InputControl(name = "leftStickPress", template = "Button", bit = (uint)Button.LeftStick)]
+        [InputControl(name = "rightStickPress", template = "Button", bit = (uint)Button.RightStick)]
+        [InputControl(name = "leftShoulder", template = "Button", bit = (uint)Button.LeftShoulder)]
+        [InputControl(name = "rightShoulder", template = "Button", bit = (uint)Button.RightShoulder)]
+        [InputControl(name = "leftTrigger", template = "Button", bit = (uint)Button.LeftTrigger)]
+        [InputControl(name = "rightTrigger", template = "Button", bit = (uint)Button.RightTrigger)]
+        public uint buttons;
+        public fixed float buttonValues[kMaxButtons];

```

Comment:
What's this for?

Reply:
Like mentioned buttons on iOS are returned as bool and float float, so the plan is to add additional InputControls later, for ex, leftShoulderRaw, XRaw returning float values.

Reply:
Wondering if it doesn't make more sense to directly wire all the buttons to the float values then and ignore the bit values. ButtonControls are AxisControls and are fine having float values.

Reply:
I talked to Mantas regarding this... and he said, that OS might have a different logic for defining when button bool value is true or false... So I slowly backed away from this solution